cfme/utils/appliance/__init__.py

```diff
@@ -2726,6 +2728,26 @@ def __init__(self, appliance=None):
         self.appliance = appliance or get_or_create_current_appliance()
 
 
+class ObjectCollections(ApplianceCollections):
+    def __init__(self, parent):
+        self._collection_cache = {}
+        self.parent = parent
+        self.appliance = self.parent.appliance
+        self.collections = self.parent._collections
+        self.load_collections()
+
+    def load_collections(self):
+        for collection, cls_and_or_filter in self.collections.items():
+            filter = {'parent': self.parent}
+            if isinstance(cls_and_or_filter, tuple):
+                filter.update(cls_and_or_filter[1])
+                cls = cls_and_or_filter[0]
+            else:
+                cls = cls_and_or_filter
+            collection_instance = cls(self.appliance, filters=filter)
+            self._collection_cache[collection] = collection_instance
```

Comment:
we should put that cache up for review

we should have a design review brainstorming around this as its potentially something that should be done in sentaku collections after we figure out what we need